// Copyright 2020 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style
// license that can be found in the LICENSE file or at
// https://developers.google.com/open-source/licenses/bsd

// This file defines protobufs for users and related business
// objects, e.g., users, user preferences.

syntax = "proto3";

package monorail.v1;

import "google/api/resource.proto";
import "google/api/field_behavior.proto";

// User represents a user of the Monorail site.
// Next available tag: 6
message User {
  // Resource name of the user.
  // Format: users/{user_id}
  string name = 1;
  // Obscured or un-obscured user email or name to show other users using the site.
  string display_name = 2 [ (google.api.field_behavior) = OUTPUT_ONLY ];
  // The global site role for a User.
  // Next available tag: 3
  enum SiteRole {
    // Default value. This value is unused.
    SITE_ROLE_UNSPECIFIED = 0;
    // Normal site user with no special site-wide extra permissions.
    NORMAL = 1;
    // Site-wide admin role.
    ADMIN = 2;
  }
  SiteRole site_role = 3;
  // User-written indication of their availability or working hours.
  string availability_message = 4;
  // Resource name of a linked primary User.
  string linked_primary_user = 5 [
      (google.api.resource_reference) = {type: "api.crbug.com/User"},
      (google.api.field_behavior) = OUTPUT_ONLY ];
}